### DISCUSSION TITLE
Broken FormItem

### DISCUSSION THREAD

**Discussion by reqyou** [UI5/webcomponents-react#4483] [Q&A] [answered]
Hey everyone,
I just noticed that a wrapped FormItem inside the default Form isn't styled properly. Its missing the default FlexBox behavior. Does anyone why this is happening? Is this intended?

I've attached a demo.

[codesandbox](https://codesandbox.io/p/sandbox/quirky-blackwell-dhd48aasd-forked-lon88y?selection=%5B%7B%22endColumn%22%3A10%2C%22endLineNumber%22%3A9%2C%22startColumn%22%3A10%2C%22startLineNumber%22%3A9%7D%5D&file=%2Fsrc%2FApp.tsx)

Best,
req

**Comment by Lukas742** [comment]
Hi @reqyou 

using components other than FormItem or FormGroup as `children` is not supported by the Form component.

**Comment by Lukas742** [comment, marked as answer]
Hi @reqyou 

good news, from[ v1.17.0](https://github.com/SAP/ui5-webcomponents-react/releases/tag/v1.17.0) on we support custom components as children.

--> [codeSandbox](https://codesandbox.io/p/sandbox/quirky-blackwell-dhd48aasd-forked-87mjm2?file=%2Fsrc%2FApp.tsx%3A1%2C1)

**Reply by reqyou** [reply]
Love this!!!

Two ideas:
- I would like to set `direction` of `FormItem`, to be able to decide whether my label is shown `column` or `row` wise
- `Button` should have `form` attribute to be able to submit a form by unique id from "outside", in case my submit button is not the nearest submit button to the form (currently we need to submit those forms manually the imperative way)

**Reply by Lukas742** [reply]
If I understand what you want to achieve correctly, then this behavior should already be possible.

- You can control the position and size of your label by setting the `labelSpanXYZ` prop. If you want it to take up a whole row, then you can set it to 12.

- For submitting via button (or Enter) only, you need to set the `Button` component to the respective `type` and nest it and the `Form` component inside a custom `form` tag. Since by default, there would be two nested forms which is not supported, you need to set the `as` prop to a `"div"` for example.

--> [codeSandbox](https://codesandbox.io/p/sandbox/quirky-blackwell-dhd48aasd-forked-4kh3hs?file=%2Fsrc%2FApp.tsx%3A39%2C62)

**Reply by reqyou** [reply]
Until now .. I misunderstood the `labelSpan` props. Thanks for that!! 💛 

About the second I adjusted the codesandbox to point out what I mean --> [codesandbox](https://codesandbox.io/p/sandbox/quirky-blackwell-dhd48aasd-forked-zmdsxn?file=%2Fsrc%2FApp.tsx%3A41%2C30)

**Reply by Lukas742** [reply]
>About the second I adjusted the codesandbox to point out what I mean --> [codesandbox](https://codesandbox.io/p/sandbox/quirky-blackwell-dhd48aasd-forked-zmdsxn?file=%2Fsrc%2FApp.tsx%3A41%2C30)

This sounds like a legitimate feature request to me. Since the`Button` component is a web component developed by the UI5 web component team, feel free to open a [feature request](https://github.com/SAP/ui5-webcomponents/issues/new/choose) in their repo and ask for the `form` attribute to be provided.
